src/components/ConditionsLayer/Flow/Flow.tsx

```diff
@@ -0,0 +1,235 @@
+import { MouseEvent, useCallback, useEffect, useState } from "react";
+import ReactFlow, {
+  Background,
+  Controls,
+  Edge,
+  Node,
+} from "react-flow-renderer";
+import ConditionNode, {
+  ConditionNodeTypes,
+} from "components/ConditionsLayer/Nodes/ConditionNode";
+import DashedEdge from "components/Edges/DashedEdge";
+import {
+  EdgeTypes,
+  initialEdges,
+  initialNodes,
+  NodeTypes,
+} from "./initialElements";
+import StartEndNode from "../Nodes/StartEndNode";
+import { StartEndNodeTypes } from "../Nodes/StartEndNode/StartEndNode";
+
+const nodeTypes = { condition: ConditionNode, startEnd: StartEndNode };
+const edgeTypes = {
+  dashed: DashedEdge,
+};
+
+interface FlowProps {
+  arePlaceholdersVisible: boolean;
+}
+
+const findLastColumn = (nodes: Node[]) => {
+  let lastCol = 0;
+  nodes.map(({ data }) => (data.col > lastCol ? (lastCol = data.col) : null));
```

Comment:
suggestion: this could be replaced with a for loop, some sort function, or even `.reduce`, but I'll leave it up yo you since it mostly impacts readibility

Reply:
Maybe even something like `Math.max(...nodes.map(({data}) => data.col)) could work?